### PR TITLE
Use sha1, instead of md5

### PR DESCRIPTION
**Pull request by szubster**
This allows prebuild-install to be used on fips systems

### CODE REVIEW

**Comment by pasieronen** [comment]
Just ran into the same issue this week - your fix looks good to me!

**Comment by karolkalinski** [comment]
I have had the same issue when working with sharp-js library. Is there a chnace that this PR would be approved? cc: @vweevers

**Comment by krethh** [comment]
@lovell Any chance we could get this merged? I'm also running into the me issue. FIPS-compliant systems ban the use of MD5 altogether so any libraries which have `prebuild-install` as a dependency will not build.

**Comment by lovell** [comment]
For those using `sharp` please upgrade as the latest version no longer depends on `prebuild-install`.

File: util.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ function trimSlashes (str) {
 }
 
 function cachedPrebuild (url) {
-  const digest = crypto.createHash('md5').update(url).digest('hex').slice(0, 6)
+  const digest = crypto.createHash('sha1').update(url).digest('hex').slice(0, 6)
```
Comment (lovell) [inline util.js:77]: I'd probably go with `sha512` here as that's what (p)npm use for content-based hashing.

Comment (szubster) [reply util.js]: done

File: util.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ function trimSlashes (str) {
 }
 
 function cachedPrebuild (url) {
-  const digest = crypto.createHash('md5').update(url).digest('hex').slice(0, 6)
+  const digest = crypto.createHash('sha512').update(url).digest('hex').slice(0, 6)
```
Comment (lovell) [inline util.js:77]: If the reason for not using an MD5 hash under the rules of FIPS is its predictability and therefore increased chance of collision then the subsequent use of `slice(0, 6)` negates hash choice anyway. :sweat_smile:

**Review by lovell — APPROVED** [review]

File: util.js (hunk 2 above)
Comment (szubster) [reply util.js:77]: Well, it doesn't much matter in the end. This code will not be running as part of the system, only during build.
At the same time, node unfortunately, does not have an equivalent of `usedforsecurity` parameter from python's [hashlib](https://docs.python.org/3/library/hashlib.html)

**Comment by szubster** [comment]
Ready to merge? :)

**Comment by lovell** [comment]
7.1.2 now available

**Pull request merged**